[CI] Add workflow_dispatch.inputs to control dashboard runs

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #101279

Summary: This gives a finer control for developers to specify which set
of configs to measure for their one-off dashboard run. Right now the
queuing for those runs look pretty bad.

Another change here is reducing the inference measurement frequency to
2 times a week.

desertfire committed:
[CI] Add workflow_dispatch.inputs to control dashboard runs

Summary: This gives a finer control for developers to specify which set
of configs to measure for their one-off dashboard run. Right now the
queuing for those runs look pretty bad.

Another change here is reducing the inference measurement frequency to
2 times a week.

[ghstack-poisoned]

desertfire committed:
Update on "[CI] Add workflow_dispatch.inputs to control dashboard runs"


Summary: This gives a finer control for developers to specify which set
of configs to measure for their one-off dashboard run. Right now the
queuing for those runs look pretty bad.

Another change here is reducing the inference measurement frequency to
2 times a week.

[ghstack-poisoned]

huydhn commented on .github/workflows/inductor-perf-test-nightly.yml:
```diff
@@ -33,13 +63,40 @@ jobs:
         ]}
 
-  linux-bionic-cuda11_8-py3_10-gcc7-inductor-test:
+  linux-bionic-cuda11_8-py3_10-gcc7-inductor-test-5days:
```
I thought this means run every 5 days, but I realize now that it means run 5 times per day.  Please make sure that the job could finish within the hour from what I see.  Otherwise, I think the latter job has the chance of cancelling the previous running one I think (if they base on the same SHA commit)

desertfire replied:
Oh, no, I got the string format wrong. Meant to run it 5 days a week (and 2 days a week for the other one). Let me fix it.

huydhn commented on .github/workflows/inductor-perf-test-nightly.yml:
```diff
@@ -33,13 +63,40 @@ jobs:
     with:
-      build-environment: linux-bionic-cuda11.8-py3.10-gcc7-sm80
+      # training only
+      build-environment: linux-bionic-cuda11.8-py3.10-gcc7-sm80-training-true-inference-false-default-true-dynamic-true-cudagraphs-true
```
This is wicked, the way you pass the parameters into test.sh via `build-environment`

desertfire committed:
Update on "[CI] Add workflow_dispatch.inputs to control dashboard runs"


Summary: This gives a finer control for developers to specify which set
of configs to measure for their one-off dashboard run. Right now the
queuing for those runs look pretty bad.

Another change here is reducing the inference measurement frequency to
2 times a week.

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "only affects dashboard run"
